starter-code/app.js

```diff
@@ -42,6 +61,9 @@ app.use(function(err, req, res, next) {
   res.locals.error = req.app.get("env") === "development" ? err : {};
 
   // render the error page
+  console.log("UNEXPECTED ERROR", req.method, req.path, err);
+
+  //if (res not sent)
```

Comment:
missing  the check for headers sent

with this code, if a middleware generates an error AFTER ending the response - `res.render(...)` - or redirecting - `res.redirect(...)` - then this middleware is going to attempt to send an error page and it will FAIL to do so